File: frontend/user/lib/data/mapper/product/product_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/mapper/product/product_detail_mapper.dart';
import 'package:b2205946_duonghuuluan_luanvan/data/mapper/product/product_image_mapper.dart';
import 'package:b2205946_duonghuuluan_luanvan/data/models/product/product_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/product/product.dart';

class ProductMapper {
  static Product fromModel(ProductModel model) {
    return Product(
      id: model.id,
      name: model.name,
      description: model.description ?? "",
      unit: model.unit ?? "",
      categoryId: model.categoryId,
      images: (model.productImages ?? [])
          .map(ProductImageMapper.fromModel)
          .toList(),
      designViews: (model.designViews ?? [])
          .map(ProductImageMapper.fromModel)
          .toList(),
      productDetails: (model.productDetails ?? [])
          .map(ProductDetailMapper.fromModel)
          .toList(),
      createdAt: model.createdAt,
      updatedAt: model.updatedAt,
    );
  }
}

File: backend/app/infrastructure/database/mappers/product_mapper.py
from app.domain.entities.product_entity import ProductEntity
from app.infrastructure.database.models.product import Product


class ProductMapper:
    @staticmethod
    def to_entity(model: Product) -> ProductEntity:
        return ProductEntity(
            id=model.id,
            category_id=model.category_id,
            name=model.name,
            description=model.description,
            unit=model.unit.value if hasattr(model.unit, 'value') else model.unit,
            created_at=model.created_at,
            updated_at=model.updated_at,
        )

    @staticmethod
    def to_model(entity: ProductEntity) -> Product:
        return Product(
            id=entity.id,
            category_id=entity.category_id,
            name=entity.name,
            description=entity.description,
            unit=entity.unit,
        )
